### PR TITLE
DFA for digits

### PR DESCRIPTION
**Pull request by btruhand**
Well as title says

### CODE REVIEW

File: include/digitsDFA.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef __DIGITSDFA_H__
+#define __DIGITSDFA_H__
+
+#include "dfa.h"
+#include <string>
+
+int error(char c); // used in error state
```
Comment (ericluii) [inline include/digitsDFA.h:7]: Don't need to include this lines 7-9 in header. They won't and shouldn't be used by anyone else.

File: include/digitsDFA.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+int checkNonZero(char c); // check for non zero digits at start state
+int checkDigits(char c); // check for digits in accepting state
+
+class DigitsDFA : public Dfa {		
+	void initDFA();
```
Comment (ericluii) [inline include/digitsDFA.h:12]: Should be initDfa not initDFA -- also put private: above

File: include/digitsDFA.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#define __DIGITSDFA_H__
+
+#include "dfa.h"
+#include <string>
+
+int error(char c); // used in error state
+int checkNonZero(char c); // check for non zero digits at start state
+int checkDigits(char c); // check for digits in accepting state
+
+class DigitsDFA : public Dfa {
```
Comment (ericluii) [inline include/digitsDFA.h:11]: I think ValidNumberDfa or something of that sorts is a better name than DigitsDFA. Makes it more clear what this DFA is trying to detect.

File: include/digitsDFA.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#include "dfa.h"
+#include <string>
+
+int error(char c); // used in error state
+int checkNonZero(char c); // check for non zero digits at start state
+int checkDigits(char c); // check for digits in accepting state
+
+class DigitsDFA : public Dfa {		
+	void initDFA();
+  public:
```
Comment (ericluii) [inline include/digitsDFA.h:13]: Indentation is off -> 4 spaces

File: include/states.h [hunk 5]
```diff
@@ -14,4 +14,16 @@ enum COMPILER_PHASE {
     CP_TOTAL_PHASES,
 };
 
+enum DFA_STATES {
```
Comment (ericluii) [inline include/states.h:17]: Can we make this and DFA_RETURN a single enumeration? also, indentation again

File: include/states.h [hunk 6]
```diff
@@ -14,4 +14,16 @@ enum COMPILER_PHASE {
...
+	DS_ERROR = 0,
+	DS_START,
+	DS_ACCEPT
```
Comment (ericluii) [inline include/states.h:20]: It's good practice to put a comma after the last value in the enumeration, so that if we add another value later, it keeps the diff clean

tl;dr; - add a comma after DS_ACCEPT

File: src/digitsDFA.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
+#include "digitsDFA.h"
+#include "states.h"
+#include <utility>
+using namespace std;
+
+int error(char c) {
+	return DS_ERROR; // always go to error
```
Comment (ericluii) [inline src/digitsDFA.cpp:7]: indentation again

File: src/digitsDFA.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+int checkDigits(char c) {
+	if('0' <= c && c <= '9') {
+		return DS_ACCEPT;
+	}
+	return DS_ERROR; // was not a digit character go to error
+}
+
+void DigitsDFA::initDFA() {
+	// initialization of the dfa
```
Comment (ericluii) [inline src/digitsDFA.cpp:25]: Maybe stick to this commenting style, instead of doing to horizontally, then the line above

File: src/digitsDFA.cpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
+#include "digitsDFA.h"
+#include "states.h"
+#include <utility>
+using namespace std;
+
+int error(char c) {
+	return DS_ERROR; // always go to error
+}
+
+int checkNonZero(char c) {
```
Comment (ericluii) [inline src/digitsDFA.cpp:10]: maybe name it as isNonZeroDigit and isDigit --> not digging the check_____ naming convention

File: src/digitsDFA.cpp [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+void DigitsDFA::initDFA() {
+	// initialization of the dfa
+	dfa[DS_ERROR] = make_pair(DR_ERROR, &error);
+	dfa[DS_START] = make_pair(DR_RUNNING, &checkNonZero);
+	dfa[DS_ACCEPT] = make_pair(DR_ACCEPT, &checkDigits);
+}
+
+DigitsDFA::DigitsDFA() : Dfa() {
+	initDFA();
```
Comment (ericluii) [inline src/digitsDFA.cpp:32]: initDfa not initDFA

**Comment by btruhand** [comment]
Cleaned up commits and creating new PR.
